### PR TITLE
ANN: Better detection of duplicate imports/items E0252

### PR DESCRIPTION
**Pull request by rrevenantt**

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 1]
```diff
@@ -92,6 +94,15 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         element.accept(visitor)
     }
 
+    private fun checkUseSpeck(holder: AnnotationHolder, useSpeck: RsUseSpeck) {
+        if (useSpeck.isStarImport || useSpeck.useGroup != null) return
+        val duplicates = holder.currentAnnotationSession.duplicatesByNamespace(useSpeck.containingMod, false)
+        if (useSpeck.namespaces.any { useSpeck in duplicates[it].orEmpty() }) {
+            val identifier = PsiTreeUtil.getDeepestLast(useSpeck)
```
[the remark this answers is not in the document]
Comment (rrevenantt) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Cargo check outputs message to console so it has to include some context. In IDE we see context right there, so i think it is better to display most specific error element.

Also for duplicate function definition intellij-rust is currently highlighting only function name while cargo check underscores whole function declaration.

File: src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt [hunk 2]
```diff
@@ -870,7 +902,7 @@ private fun AnnotationSession.duplicatesByNamespace(owner: PsiElement, recursive
     return duplicates
 }
 
-private fun PsiElement.namedChildren(recursively: Boolean, stopAt: Class<*>? = null): Sequence<RsNamedElement> {
+private fun PsiElement.namedChildren(recursively: Boolean, stopAt: Class<*>? = null): Sequence<PsiElement> {
```
[the remark this answers is not in the document]
Comment (rrevenantt) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: It is necessary to be able to concatenate its result with `Sequence<RsUseSpec>`.

[the remark this answers is not in the document]
Comment (mchernyavsky) [reply src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt]: Hmm, I have tried to rollback the line, and the code compiles just fine.

**Comment by mchernyavsky** [comment]
bors retry

**Pull request merged**